main_test.go

```diff
@@ -200,3 +200,23 @@ func FuzzGetTokenName(f *testing.F) {
 		assert.Equal(t, strings.ToLower(orig), tokenName, "URL: "+urlString+"URL String: "+url.String()+" Scheme: "+url.Scheme)
 	})
 }
+
+func TestParseJobAd(t *testing.T) {
+	// Job ad file does not exist
+	err := os.WriteFile(".job.ad", []byte("Owner = \"john\"\nProjectName = \"test-project\""), 0644)
```

Comment:
I think you can create a temporary file with the testing.T struct, [TempDir](https://pkg.go.dev/testing#T.TempDir).  Then write to that directory and set the _CONDOR_JOB_AD environment variable to point to it.  Then you don't have to worry about clean up, and won't pollute the current working directory.